google/cloud/pubsub/internal/tracing_message_batch.h

```diff
@@ -30,19 +28,19 @@ namespace cloud {
 namespace pubsub_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
-class TracingMessageBatchPeer;
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
 
 /**
  * Records spans related to a batch messages across calls and
  * callbacks in the `BatchingPublisherConnection`.
  */
 class TracingMessageBatch : public MessageBatch {
  public:
-  explicit TracingMessageBatch(std::unique_ptr<MessageBatch> child)
+  explicit TracingMessageBatch(std::shared_ptr<MessageBatch> child)
```

Comment:
Do you still need this class in the header file?  Meaning: can you rewrite the tests to just use the `MakeTracingMessageBatch()` function?

If you can just use the function, the `#includes` and `#ifdef` blocks could get simpler.